synapse/handlers/sync.py

```diff
@@ -1603,11 +1610,16 @@ async def _generate_sync_entry_for_to_device(
             )
 
             for message in messages:
-                # We pop here as we shouldn't be sending the message ID down
-                # `/sync`
-                message_id = message.pop("message_id", None)
-                if message_id:
-                    set_tag(SynapseTags.TO_DEVICE_MESSAGE_ID, message_id)
+                log_kv(
+                    {
+                        "event": "to_device_message",
+                        "sender": message["sender"],
+                        "type": message["type"],
+                        EventContentFields.TO_DEVICE_MSGID: message["content"].get(
```

Comment:
Was this supposed to be `SynapseTags.TO_DEVICE_MSGID` instead of `EventContentFields.TO_DEVICE_MSGID`?

Reply:
not really, no. Note that this is being used in `log_kv`, which adds entries to the "logs" section of the span:

![image](https://user-images.githubusercontent.com/1389908/210361735-304aa06e-4d75-466c-94c1-02e4a83b86e1.png)

The "logs" are not searchable, so are basically free text. I figured it made more sense to make the log fields match the fields on the wire.

`SynapseTags.*` are intended for use with `set_tag`, which adds entries to the "tags" section of the span:

![image](https://user-images.githubusercontent.com/1389908/210362054-93c8fbf9-cf25-440e-b1d2-16a72cc6eff2.png)

Since these *are* searchable (that's the whole point of them), it seemed to make sense to use tags that were namespaced with a `to_device` prefix.

Reply:
That makes sense, thanks for the explanation!